packages/gatsby-reporter/src/reporter.ts

```diff
@@ -0,0 +1,260 @@
+import { stripIndent } from "common-tags"
+import chalk from "chalk"
+import { trackError } from "gatsby-telemetry"
+import { globalTracer, Span } from "opentracing"
+import { getErrorFormatter } from "./get-error-formater"
+
+import * as reporterActions from "./redux/actions"
+import { LogLevels, ActivityStatuses } from "./constants"
+import constructError from "./structured-errors/construct-error"
+import { prematureEnd } from "./catch-exit-signals"
+import { IStructuredError } from "./structured-errors/types"
+import { createTimerReporter, ITimerReporter } from "./reporter-timer"
+import { createPhantomReporter, IPhantomReporter } from "./reporter-phantom"
+import { createProgressReporter, IProgressReporter } from "./reporter-progress"
+import { ErrorMeta, CreateLogAction } from "./types"
+
+const errorFormatter = getErrorFormatter()
+const tracer = globalTracer()
+
+export interface IActivityArgs {
+  id?: string
+  parentSpan?: Span
+  tags?: { [key: string]: any }
+}
+
+let isVerbose = false
+
+/**
+ * Reporter module.
+ * @module reporter
+ */
+class Reporter {
```

Comment:
It would be nice to export this as just a type without exposing it as a class that people can instantiate.

Reply:
This isn't part of the public API. They would have to reach in and grab it. Which is always a risk in JS land. But the actual export is a single instance of this. Make sense?

Reply:
We need the type because at the moment `reporter` is passed around in several places, and the type is exported from `gatsby` (which I guess means it is public API). We're having to type things as `typeof reporter`, which isn't ideal. We could just refactor it out everywhere, because it doesn't really make sense to pass it around when it's a singleton, but as it stands it's used quite a lot.

Reply:
Yeah agreed. I updated the lib to export the typeof the reporter on `IReporter`. Ooo I should change it to `IGatsbyReporter`